### PR TITLE
Update extension workflows and docs

### PR DESCRIPTION
**Pull request by phnzb**
- added reusable workflows for extensions:
  - `manifest.yml` - check extension manifest
  - `prospector.yml` - run prospector static analysis
- added `check-master-manifest.yml` workflow to check masterl list (`extensions.json`) on PR and feature/* branches
- updated README.md with extension flow

### CODE REVIEW

**Review by dnzbk — DISMISSED** [review]

File: README.md [hunk 1]
```diff
@@ -7,13 +7,33 @@ This repository contains a list of NZBGet extensions, which is used by the Exten
 1. Make your extension compatible with NZBGet spec v23+. 
 [Documentation](https://github.com/nzbgetcom/nzbget/blob/develop/docs/extensions/EXTENSIONS.md).
 
-2. Once your extension is ready, please contact NZBGet maintainers to get it added into the extension manager 
-(we'll create a project for you - `https://github.com/nzbgetcom/Extension-YourExtensionName/`).
+2. Make sure that yours extension repository contains the required and possibly optional workflows:
```
Comment (luckedea) [inline README.md:10]: typo, your

File: README.md [hunk 2]
```diff
@@ -7,13 +7,33 @@ This repository contains a list of NZBGet extensions, which is used by the Exten
...
 
-3. Create a fork of that repository, add your extension to it, and make a Pull request.
+Mandatory:
+- Prospector Python Static Analysis check
+- Manifest check
 
->You can use any of nzbgetcom extension as example, 
-e.g [FakeDetector](https://github.com/nzbgetcom/Extension-FakeDetector).
+Optional:
+- Tests check (availability of tests is highly recommended)
```
Comment (luckedea) [inline README.md:17]: (availability of tests is highly recommended) rephrase this, reads a bit odd for me.

File: README.md [hunk 3]
```diff
@@ -7,13 +7,33 @@ This repository contains a list of NZBGet extensions, which is used by the Exten
...
+- Release workflow
+
+Reusable workflows can be used from the current repository. For an example of their use, please refer to the Example extensions repository (see below).
+
+3. Once your extension is ready, please contact NZBGet maintainers to get it added into the extension manager.
```
Comment (luckedea) [inline README.md:22]: Contact how?

File: README.md (hunk 2 above)
Comment (phnzb) [reply README.md]: -> Having tests for your extension is highly recommended

File: README.md (hunk 3 above)
Comment (phnzb) [reply README.md]: (by creating Issue or Discussion)

**Review by dnzbk — APPROVED** [review]

**Pull request merged**